test/test_utils/ColdResumeManager.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#include "test/test_utils/ColdResumeManager.h"
+
+namespace rsocket {
+
+ColdResumeManager::~ColdResumeManager() {}
+
+void ColdResumeManager::trackReceivedFrame(
+    size_t frameLength,
+    FrameType frameType,
+    StreamId streamId,
+    size_t consumerAllowance) {
+  if (!shouldTrackFrame(frameType)) {
+    return;
+  }
+  auto it = streamResumeInfos_.find(streamId);
+  // If streamId is not present in streamResumeInfo it likely means a
+  // COMPLETE/CANCEL/ERROR was received in this frame and the
+  // ResumeMananger::onCloseStream() was already invoked resulting in the entry
+  // being deleted.
+  if (it != streamResumeInfos_.end()) {
+    it->second.consumerAllowance = consumerAllowance;
+  }
+  WarmResumeManager::trackReceivedFrame(
+      frameLength, frameType, streamId, consumerAllowance);
+}
+
+void ColdResumeManager::trackSentFrame(
+    const folly::IOBuf& serializedFrame,
+    FrameType frameType,
+    folly::Optional<StreamId> streamIdPtr,
+    size_t consumerAllowance) {
+  if (!shouldTrackFrame(frameType)) {
+    return;
+  }
+  const StreamId streamId = *streamIdPtr;
```

Comment:
Why don't we need to check if `streamIdPtr` is `folly::none`?

Reply:
All frames should have StreamID.  So I want to get rid of the `folly::Optional` part for the StreamID in another PR.  For now, will add the check.